### PR TITLE
Surge Alert Status and Ordering based on status and opens fields

### PR DESCRIPTION
**Pull request by susilnem**
## Addresses
- https://github.com/IFRCGo/go-api/issues/2036

## Changes
- Add Status Field in Alert
- Add status in enums
- Create Cronjob Command to update the Alert status

## Post deployment actions
@szabozoltan69 We need to run this after deployment to update the status. After that, it will be updated every 12 hours.
```
./manage.py update_surge_alert_status
```

## Checklist
Things that should succeed before merging.

- [ ] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

**Review by k9845 — APPROVED** [review]
lets add test case as well

**Review by k9845 — COMMENTED** [review]
@susilnem Lets also add test case for this

File: notifications/models.py [hunk 1]
```diff
@@ -23,6 +23,10 @@ class SurgeAlertCategory(models.IntegerChoices):
     SHELTER = 3, _('shelter')
     STAND_DOWN = 4, _('stand down')
 
+class SurgeAlertStatus(models.IntegerChoices):
```
Comment (thenav56) [inline notifications/models.py:26]: Let's add a note, explaining the ordering usage here

File: notifications/tests.py [hunk 2]
```diff
@@ -136,3 +139,40 @@ def _to_csv(*items):
...
+            country=country_2,
+            molnix_tags=[molnix_tag_1, molnix_tag_3],
+            opens = timezone.now() - timedelta(days=2),
+            closes = timezone.now() - timedelta(days=1)
+
+        )
+        self.assertEqual(alert1.status, SurgeAlertStatus.OPEN)
+        self.assertEqual(alert2.status, SurgeAlertStatus.CLOSED)
+
+        time.sleep(10)
```
Comment (thenav56) [inline notifications/tests.py:171]: Hmm... let's not do this… Let's mock the date time if needed.

File: notifications/management/commands/update_alert_status.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                status=models.Case(
+                    models.When(is_stood_down=True, then=models.Value(SurgeAlertStatus.STOOD_DOWN)),
+                    models.When(closes__lt=now, then=models.Value(SurgeAlertStatus.CLOSED)),
+                    models.When(closes__gte=now, then=models.Value(SurgeAlertStatus.OPEN)),
+                    default=models.F('status'),
+                    output_field=models.IntegerField()
+                )
+            )
+        except Exception as e:
+            logger.error('Error while updating alerts status: %s' % str(e))
```
Comment (thenav56) [inline notifications/management/commands/update_alert_status.py:31]: Let's use exc_info=True
```suggestion
            logger.error('Error while updating alerts status', exc_info=True)
```

File: notifications/management/commands/update_alert_status.py
Comment (thenav56) [inline notifications/management/commands/update_alert_status.py:1]: Let's not use generic names for stuffs which are defined as global

Maybe `update_surge_alert_status`

File: notifications/models.py [hunk 4]
```diff
@@ -54,6 +58,7 @@ class SurgeAlert(models.Model):
 
     # Don't set `auto_now_add` so we can modify it on save
     created_at = models.DateTimeField(verbose_name=_('created at'))
+    status = models.IntegerField(choices=SurgeAlertStatus.choices, verbose_name=_('alert status'), default=0)
```
Comment (thenav56) [inline notifications/models.py:61]:
```suggestion
    status = models.IntegerField(choices=SurgeAlertStatus.choices, verbose_name=_('alert status'), default=SurgeAlertStatus.OPEN)
```

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. Few comments

**Review by thenav56 — APPROVED** [review]

**Comment by thenav56** [comment]
Hey @szabozoltan69, could you review and merge/deploy this to staging?

**Comment by thenav56** [comment]
Hey @szabozoltan69 

We also need to run this after deployment to initiate the status. After that, it will be updated every 12 hours.


./manage.py update_surge_alert_status

**Comment by szabozoltan69** [comment]
Hey @thenav56 
Deployment was ok, also the migration:
Running migrations:
  Applying notifications.0014_surgealert_status... OK
Also this was run flawless:
./manage.py update_surge_alert_status

**Comment by k9845** [comment]
@szabozoltan69 Can you run `./manage.py update_surge_alert_status` command on production? I think we missed this.

**Comment by szabozoltan69** [comment]
Now I did it.

**Pull request merged**